### ISSUE TITLE
Can't get the plugin to work with Nuxt and Vitest

### ISSUE THREAD

**Issue by datsenkoboos** [aleclarson/vite-tsconfig-paths#123]
Hi, i'm trying to use **Vitest** and **Nuxt 3** with its "explicit" imports from `'#imports'` which is declared in `tsconfig.json`:
`tsconfig.json` in the root of the project
```
{
  // https://nuxt.com/docs/guide/concepts/typescript
  "extends": "./.nuxt/tsconfig.json"
}
```
`"./.nuxt/tsconfig.json"`
```
{
  "compilerOptions": {
    ...
    "paths": {
      ...
      "#imports": [
        "C:/path/to/project/nuxt-build/.nuxt/imports"
      ...
    }
   ...
  },
```
Vitest can't resolve the import even with `vite-tsconfig-paths`:

![изображение](https://github.com/aleclarson/vite-tsconfig-paths/assets/88643277/b37e32bf-d827-45b4-8f93-97a3bfaa6f43)

`vite.config.ts`
```
import { defineConfig } from 'vite'
import tsconfigPaths from 'vite-tsconfig-paths'

export default defineConfig({
  plugins: [tsconfigPaths()],
})
```
Am i doing anything wrong?

**Comment by aleclarson** [comment]
I'm not familiar with Nuxt. Why do they use `#imports` specifically? I suppose this plugin is tripping on the `#` character somewhere, but haven't tested it myself. PR welcome

**Comment by akasection** [comment]
Hello
I also encountered this issue and seems to find the root cause (though I'm unsure how to fix that)

A bit more explanation:

So basically Nuxt dev used custom paths like `~/`, `#imports` etc. to allow importing some nuxt framework functions into Vue files.
To make the intellisense working with this behaviour, the Nuxt adds the resolve alias and generate custom tsconfig with relative paths under `<rootDir>/.nuxt/tsconfig.json` 

So in most common Nuxt project, we will have two tsconfig:
1. `<rootDir>/tsconfig.json` (which extends below)
2. `<rootDir/.nuxt/tsconfig.json`

Later in nuxt update, every path aliases on the .nuxt tsconfig now has relative paths with added `../` (up a folder), which previously it was absolute paths. Though in actual development, there is no behaviour change at all. Most Vue files can import things from `#import` just fine.

Now the problem is on Vitest, using `vite-tsconfig-paths`, somehow, the plugin will resolve the tsconfig alias `#import` into.. let's say, `../node_modules/nuxt/dist/app` , but it doesn't account that the tsconfig position was on `.nuxt`, but rather resolved relatively to `rootDir` instead.

So, absolutely, vite-tsconfig-paths will resolve any path aliases of `.nuxt/tsconfig.json` into like `<rootDir>/../<things>` which is parent folder of the project itself.

Overall, it's not just the `#imports` alias that broken, but all aliases (including the `~` alias, `@` and any relative aliases. The absolute pathes are unaffected.

in summary, currently any tsconfig that has relative paths AND placed inside subfolders of root project will misbehave:
```json
// This is broken
"paths": {
      "~": [
        ".."
      ],
      "~/*": [
        "../*"
      ],
     "assets": [
        "../assets"
      ],
      "public": [
        "../public"
      ],
      "public/*": [
        "../public/*"
      ],
}
```

**Comment by JasonLandbridge** [comment]
@datsenkoboos or @akasection, did you guys ever managed to find a solution for this?

**Comment by JasonLandbridge** [comment]
This might be related to this issue here: https://github.com/nuxt/nuxt/issues/26512

It seems nuxt changed the way their `tsconfig.json` are generated which might lead to non-working extends

**Comment by LiNCH35** [comment]
For me it also helped to downgrade the `vite` version to `4.3.9`

**Comment by danielroe** [comment]
Here's an example of how to do the resolution:

https://github.com/nuxt/module-builder/blob/main/src/commands/build.ts#L279-L295

**Comment by JasonLandbridge** [comment]
@akasection Thank you very much! This indeed fixed my issue!

**Comment by aleclarson** [comment]
I believe this is a duplicate of #150, which I just opened an upstream PR for today.

**Comment by aleclarson** [comment]
Please give v5.1.4 a try and let me know if it works. If you're still experiencing an issue with Nuxt, please provide a minimal reproduction. You can do this via StackBlitz or by forking this repository and editing the `./demo` folder as necessary.
